app/helpers/application_helper/button/vm_instance_scan.rb

```diff
@@ -3,14 +3,12 @@ class ApplicationHelper::Button::VmInstanceScan < ApplicationHelper::Button::Bas
 
   def skip?
     return false if @record.kind_of?(OrchestrationStack) && @display == "instances"
-    !(@record.is_available?(:smartstate_analysis) ||
-      @record.is_available_now_error_message(:smartstate_analysis)) ||
-    !@record.has_proxy?
+    @record.archived? || !@record.has_proxy?
```

Comment:
The earlier code was `@record.is_available?(:smartstate_analysis) || @record.is_available_now_error_message(:smartstate_analysis)`

However as you can see, this OR condition always turned up to be true because in case is_available returns true, it is true. But if it is false, then is_available_now_error_message will return a reason which will again make this OR true.

So effectively, the negation of OR (always false) was something that didn't affect anything, hence it was independent of whether smartstate_analysis was supported or not.

But after removing this OR completely from the code, I found out that two specs were failing because the tests depend on whether the record is archived or not. And this was handled in earlier code by returning reason of not supporting smartstate_analysis as nil, and is_available as false.

So effectively, the specs depend only on term archived? and not whether smartstate_analysis is supported or not.

Ref: https://github.com/ManageIQ/manageiq/blob/master/spec/helpers/application_helper/buttons/vm_instance_scan_spec.rb#L27